### DISCUSSION TITLE
FzfLua lsp_workspace_symbols -- Filter out certain directories

### DISCUSSION THREAD

**Discussion by nik-rev** [ibhagwan/fzf-lua#1486] [Q&A] [answered]
Hi, thank you for your awesome plugin. I really enjoy using it. I have a question, When I run `FzfLua lsp_workspace_symbols` it also searches in some paths which I would like to ignore (namely, any files within  `.next/*` or `.contentlayer/*`. How could I do this?

I took notice how FzfLua won't look in `node_modules` by default. So I assumed there is some kind of configuration I can set to filter for other folders.

I didn't find any config option like that. is this possible to do?

![2024-10-10_20-18](https://github.com/user-attachments/assets/4ac8a10f-dd3b-4bce-b6f1-24241118ee90)

**Comment by ibhagwan** [comment, marked as answer]
There are a few ways to filter out symbols:
> Name based regex
```lua
lua require("fzf-lua").lsp_workspace_symbols({ regex_filter = {"vim",exclude=true} })
```

> Current directory only
```lua
:FzfLua lsp_workspace_symbols cwd_only=true
```

> Ignoring specific directory patterns (probably what you want):
```lua
:lua require("fzf-lua").lsp_workspace_symbols({ file_ignore_patterns = { "/usr", "local" } })
```

**Reply by nik-rev** [reply]
Thank you!
